iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperations.java

```diff
@@ -282,6 +311,98 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
     }
   }
 
+  /**
+   * Build a new partition spec with new schema from original pspec. The new pspec has the same
+   * partition fields as the original pspec with source ids from the new schema
+   *
+   * @param originalPspec
+   * @param originalSchema
+   * @param newSchema
+   * @return new partition spec
+   */
+  public static PartitionSpec rebuildPartitionSpec(
```

Comment:
these methods shouldn't be public static. 
If this is purely for testing, mocking is a better way.

Reply:
done